src/main/java/com/elytradev/concrete/inventory/gui/widget/WWidget.java

```diff
@@ -75,46 +101,94 @@ public void setParent(WPanel parent) {
-	 * drawing space is from (x, y) to (x + width - 1, y + height - 1) inclusive. However, no scissor or depth masking
-	 * is done, so please take care to respect your boundaries.
-	 * @param x The X coordinate of the leftmost pixels of this widget in device (opengl) coordinates
-	 * @param y The Y coordinate of the topmost pixels of this widget in device (opengl) coordinates
+	 * Draws the background of this Widget at the specified coordinates.
+	 * <p>
+	 * The coordinates provided are the top-level device coordinates of this
+	 * widget's topleft corner, so don't translate by the widget X/Y! That's
+	 * already been done. Your "valid" drawing space is from (x, y) to
+	 * (x + width - 1, y + height - 1) inclusive. However, no scissor or depth
+	 * masking is done, so please take care to respect your boundaries.
+	 *
+	 * @param x The X coordinate of the leftmost pixels of this widget in
+	 *          device (opengl) coordinates
+	 * @param y The Y coordinate of the topmost pixels of this widget in
+	 *          device (opengl) coordinates
+	 * @see #paintForeground
 	 */
-	public void paint(int x, int y) {
-		
-	}
+	@SideOnly(Side.CLIENT)
+	public void paintBackground(int x, int y) {}
 	
 	/**
-	 * Creates "heavyweight" component peers
-	 * @param c the top-level Container that will hold the peers
+	 * Draws the foreground of this Widget at the specified coordinates.
+	 * <p>
+	 * The coordinates provided are the top-level device coordinates of this
+	 * widget's topleft corner, so don't translate by the widget X/Y! That's
+	 * already been done. Your "valid" drawing space is from (x, y) to
+	 * (x + width - 1, y + height - 1) inclusive. However, no scissor or depth
+	 * masking is done, so please take care to respect your boundaries.
+	 *
+	 * @param x The X coordinate of the leftmost pixels of this widget in
+	 *          device (opengl) coordinates
+	 * @param y The Y coordinate of the topmost pixels of this widget in
+	 *          device (opengl) coordinates
+	 * @see #paintBackground
 	 */
-	public void createPeers(ConcreteContainer c) {
-	}
-
 	@SideOnly(Side.CLIENT)
-	public void paintBackground(int x, int y) {
-	}
+	public void paintForeground(int x, int y) {}
 	
+	/**
+	 * Determines whether this widget is valid.
+	 * <p>
+	 * A widget is valid when it is correctly sized and positioned within its
+	 * parent panel and all its children are also valid.
+	 *
+	 * @return <code>true</code> if the widget is valid, <code>false</code> otherwise
+	 * @see #validate
+	 * @see #invalidate
+	 */
 	public boolean isValid() {
 		return valid;
 	}
 	
 	/**
-	 * Creates component peers, lays out children, and initializes animation data for this Widget and all its children.
-	 * The host container must clear any heavyweight peers from its records before this method is called.
+	 * Validates this widget.
+	 * <p>
+	 * The meaning of the term <i>validating</i> is defined by the ancestors of
```

Comment:
Hm? A child class that uses inheritDoc will inherit the documentation of the parent class, not the other way around.